components/src/preact/numberSequencesOverTime/getNumberOfSequencesOverTimeTableData.ts

```diff
@@ -0,0 +1,38 @@
+import type { NumberOfSequencesDatasets } from '../../query/queryNumberOfSequencesOverTime';
+import type { TemporalGranularity } from '../../types';
+import { generateAllInRange, getMinMaxTemporal, type Temporal } from '../../utils/temporal';
+
+export const getNumberOfSequencesOverTimeTableData = (
+    data: NumberOfSequencesDatasets,
+    granularity: TemporalGranularity,
+) => {
+    const datasetsWithCountByDate = data.map(({ displayName, content }) => ({
+        displayName,
+        content: new Map(content.map((datum) => [datum.dateRange?.toString(), datum])),
+    }));
+
+    const allDateRangesThatOccurInData = datasetsWithCountByDate
+        .map(({ content }) => [...content.values()].map((datum) => datum.dateRange))
+        .reduce((acc, keys) => new Set([...acc, ...keys]), new Set<Temporal | null>());
+
+    const minMax = getMinMaxTemporal(allDateRangesThatOccurInData);
+    if (minMax === null) {
+        return [];
+    }
+
+    const allDateRanges: (Temporal | null)[] = generateAllInRange(...minMax);
```

Comment:
This should not be null for the values (from the definition of the function). When this is not the case then we should change the return type of the function.
```suggestion
    const allDateRanges: Temporal[] = generateAllInRange(...minMax);
```

Reply:
In the line below, we have to add `null`, that's why I need to allow `null` explicitly here.